File: tourHanoi.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/*
n : nombre de disques utilisés
D : emplacement de départ
A : emplacement d'arrivée
I : emplacement intermédiaire
*/

void toursHanoi(int n, char D, char A, char I)
{

    if (n == 1)
    {
        printf("Disque 1 de %c a %c \n", D, A);
        return;
    }
    else
    {
        // D à A
        toursHanoi(n - 1, D, I, A);
        printf("Disque %d de %c a %c \n", n, D, A);
        // I à A
        toursHanoi(n - 1, I, A, D);
    }
}

int main()
{
    double temps;
    clock_t start, end;
    int nDisques = 3;

    start = clock();
    toursHanoi(nDisques, 'A', 'C', 'B');
    end = clock();
    temps = (double)(end - start) / CLOCKS_PER_SEC;
    printf("Temps d'éxécution: %f secondes\n", temps);
}

File: tourHanoi.java
import java.io.*;
import java.math.*;
import java.util.*;
import java.time.Duration;

class Honoi {
    static void toursHonoi(int n, char D, char A, char I) {
        if (n == 1) {
            return;
        }
        toursHonoi(n - 1, D, I, A);
        System.out.println("Disque " + n + " de " + D + " a " + A);
        toursHonoi(n - 1, I, A, D);
    }

    public static void main(String args[]) {
        int N = 5; // NOMBRE DE DISQUE
        double start, end, time;
        start = System.currentTimeMillis();
        toursHonoi(N, 'A', 'C', 'B');
        end = System.currentTimeMillis();
        time = (end - start) / 1000F;
        System.out.println("Temps d'execution: " + time + " secondes");
    }
}